hbase-common/src/main/java/org/apache/hadoop/hbase/CellUtil.java

```diff
@@ -598,19 +598,28 @@ public static void setTimestamp(Cell cell, byte[] ts, int tsOffset) throws IOExc
   }
 
   /**
-   * @param cell
    * @return The Key portion of the passed <code>cell</code> as a String.
    */
   public static String getCellKeyAsString(Cell cell) {
-    StringBuilder sb = new StringBuilder(Bytes.toStringBinary(
-      cell.getRowArray(), cell.getRowOffset(), cell.getRowLength()));
+    return getCellKeyAsString(cell,
+      c -> Bytes.toStringBinary(c.getRowArray(), c.getRowOffset(), c.getRowLength()));
+  }
+
+  /**
+   * @param cell the cell to convert
+   * @param rowConverter used to convert the row of the cell to a string
+   * @return The Key portion of the passed <code>cell</code> as a String.
+   */
+  public static String getCellKeyAsString(Cell cell, Function<Cell, String> rowConverter) {
+    StringBuilder sb = new StringBuilder(rowConverter.apply(cell));
     sb.append('/');
-    sb.append(cell.getFamilyLength() == 0? "":
-      Bytes.toStringBinary(cell.getFamilyArray(), cell.getFamilyOffset(), cell.getFamilyLength()));
-    // KeyValue only added ':' if family is non-null.  Do same.
+    sb.append(cell.getFamilyLength() == 0 ? ""
+      : Bytes.toStringBinary(cell.getFamilyArray(), cell.getFamilyOffset(),
+        cell.getFamilyLength()));
+    // KeyValue only added ':' if family is non-null. Do same.
     if (cell.getFamilyLength() > 0) sb.append(':');
-    sb.append(cell.getQualifierLength() == 0? "":
-      Bytes.toStringBinary(cell.getQualifierArray(), cell.getQualifierOffset(),
+    sb.append(cell.getQualifierLength() == 0 ? ""
+      : Bytes.toStringBinary(cell.getQualifierArray(), cell.getQualifierOffset(),
```

Comment:
nit: I prefer the ":" on the EOL rather than these refactors. Not important.

Reply:
Let me check my formatter config...